packages/jsii-diff/bin/jsii-diff.ts

```diff
@@ -0,0 +1,131 @@
+import fs = require('fs-extra');
+import reflect = require('jsii-reflect');
+import spec = require('jsii-spec');
+import log4js = require('log4js');
+import yargs = require('yargs');
+import { compareAssemblies } from '../lib';
+import { Mismatches } from '../lib/types';
+import { downloadNpmPackage } from '../lib/util';
+import { VERSION } from '../lib/version';
+
+const LOG = log4js.getLogger('jsii-diff');
+
+async function main(): Promise<number> {
+  const argv = yargs
+      .env('JSII_DIFF')
+      .option('verbose', { alias: 'v', type: 'count', desc: 'Increase the verbosity of output', global: true })
+      .option('default-experimental', { alias: 'e', type: 'boolean', desc: 'Treat unmarked APIs as experimental', default: false })
+      .usage('$0 <original> [updated]', 'Compare two JSII assemblies.', args => args
+        .positional('original', {
+          description: 'Original assembly (file, package or "npm:package@version")',
+          type: 'string'
+        })
+        .positional('updated', {
+          description: 'New assembly (file, package or "npm:package@version")',
+          type: 'string',
+          default: '.'
+        })
+      )
+      .help()
+      .version(VERSION)
+      .argv;
+
+  configureLog4js(argv.verbose);
+
+  LOG.debug(`Loading original assembly from ${(argv as any).original}`);
+  const original = await loadAssembly((argv as any).original);
+
+  LOG.debug(`Loading updated assembly from ${(argv as any).updated}`);
+  const updated = await loadAssembly((argv as any).updated);
+
+  if (original.name !== updated.name) {
+    process.stderr.write(`Look like different assemblies: '${original.name}' vs '${updated.name}'. Comparing is probably pointless...`);
+  }
+
+  const mismatches = new Mismatches();
+
+  LOG.info(`Starting analysis`);
+  compareAssemblies(original, updated, {
+    mismatches,
+    defaultExperimental: argv["default-experimental"],
+  });
+
+  LOG.info(`Found ${mismatches.count} issues`);
+
+  if (mismatches.count > 0) {
+    process.stderr.write(`Original assembly: ${original.name}@${original.version}\n`);
+    process.stderr.write(`Updated assembly:  ${updated.name}@${updated.version}\n`);
+    process.stderr.write(`API elements with incompatible changes:\n`);
+    for (const msg of mismatches.messages()) {
+      process.stderr.write(`- ${msg}\n`);
+    }
+
+    return 1;
+  }
+
+  return 0;
+}
+
+async function loadAssembly(name: string) {
+  if (name.startsWith('npm:')) {
```

Comment:
I don't like that at all. For whatever reason I mistyped something, or my script assumes the existence of a file that wasn't properly generated by my build script, and all of a sudden I'm downloading something from the internet?

Please no.

Reply:
Hmmmmmmmkay. Then why even allow the downloading at all? Can't the user do it all themselves? What's the value in sugaring that?

Reply:
We allow the downloading because this is going to be the common case (where else are we going to get the previous interface definitions? Store them in the code repo? Please no). 

Plus, it's kinda involved, installing to a temporary directory and then erasing it afterwards. Seems like something a tool could do for me.